rasa/core/brokers/kafka.py

```diff
@@ -124,67 +127,73 @@ def publish(
                     f"Could not publish message to kafka url '{self.url}'. "
                     f"Failed with error: {e}"
                 )
-                connected = self.producer.bootstrap_connected()
-                if not connected:
-                    self._close()
+                try:
+                    self.producer.list_topics(timeout=5)
```

Comment:
i think this deserves a comment, as it isn't obvious from the code itself why this line is here (or alternatively, maybe even better, wrap this in a small function that makes the intent clear)

Reply:
Addressed: https://github.com/RasaHQ/rasa/pull/11765/commits/c34bc473b75fa51cffed3e1222c224633665f89e